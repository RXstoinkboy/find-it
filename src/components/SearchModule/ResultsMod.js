import React from 'react';
import styles from './styles/ResultsMod.module.scss'
import PropTypes from 'prop-types'

export const ResultsMod = props => {
    return (
        // display the box only if input is focused
        <div className={props.focus && props.keywordToSearch.length > 0 ? [styles.wrapper, styles.wrapperActive].join(' ') : styles.wrapper} >
        {/* display list of categories which were found from autocomplete query */}
            {Object.keys(props.autocompleteResults).map(section => (
                <section className={styles.section} key={section}>
                {/* display the title of each category but only if it is not empty */}
                    <h4 className={styles.sectionTitle}>{
                        props.autocompleteResults[section].length > 0 ?
                            section
                                : null
                    }</h4>
                    {/* display content of each category */}
                    <ul className={styles.contentList}>
                        {props.autocompleteResults[section].map((item, idx) => (
                            <li 
                                onMouseDown={props.handlePickWord}
                                value={
                                    section === 'categories' ? item.title
                                        : section === 'businesses' ? item.name
                                            : item.text
                                }
                                className={styles.contentItem} 
                                key={idx}
                            >
                                {
                                    section === 'categories' ? item.title
                                        : section === 'businesses' ? item.name
                                            : item.text
                                }
                            </li>
                        ))}
                    </ul>
                </section>
            ))}

        </div>
    );
};

export default ResultsMod;

ResultsMod.propTypes = {
    focus: PropTypes.bool.isRequired,
    keywordToSearch: PropTypes.string.isRequired,
    autocompleteResults: PropTypes.object,
    handlePickWord: PropTypes.func
}