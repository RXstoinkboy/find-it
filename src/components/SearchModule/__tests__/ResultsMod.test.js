import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import {ResultsMod} from '../ResultsMod'

it(`ResultsMod renders correctly`, ()=> {
    const props = {
        focus: false,
        keywordToSearch: 'hello',
        autocompleteResults: {cats: [], dogs: []},
        handlePickWord: jest.fn()
    }

    const resultsMod = shallow(
        <ResultsMod {...props} />
    )

    expect(toJson(resultsMod)).toMatchSnapshot()
})