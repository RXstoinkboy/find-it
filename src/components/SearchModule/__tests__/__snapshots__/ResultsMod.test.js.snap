// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsMod renders correctly 1`] = `
<div
  className="wrapper"
>
  <section
    className="section"
    key="cats"
  >
    <h4
      className="sectionTitle"
    />
    <ul
      className="contentList"
    />
  </section>
  <section
    className="section"
    key="dogs"
  >
    <h4
      className="sectionTitle"
    />
    <ul
      className="contentList"
    />
  </section>
</div>
`;
